GVRf/Extensions/3DCursor/3DCursorLibrary/src/main/java/org/gearvrf/io/cursor3d/LaserCursor.java

```diff
@@ -55,7 +57,11 @@ void dispatchSensorEvent(SensorEvent event) {
         cursorEvent.setCursorPosition(getPositionX(), getPositionY(), getPositionZ());
         cursorEvent.setCursorRotation(getRotationW(), getRotationX(), getRotationY(),
                 getRotationZ());
-        cursorEvent.setKeyEvent(event.getCursorController().getKeyEvent());
+        GVRCursorController controller = event.getCursorController();
+        cursorEvent.setMotionEvents(controller.getMotionEvents());
```

Comment:
The CursorEvent itself is not going to be valid after the callback is done. If needed the CursorEvent along with the MotionEvents need to be cloned during cursor event call back, I think I should mention this part in the javadocs. 
About thread safety, currently the model for GVRCursorController is to call update() from a single thread. Essentially having one thread for each CursorController, Gaze, Mouse and Gamepad. Making the logic simpler and avoiding the need to keep copies of the event data when not needed. Idea is that if the application logic needs, it can make copies during the callback.